index.js

```diff
@@ -8,6 +8,7 @@ function shift (stream) {
 
 function getStateLength (state) {
   if (state.buffer.length) {
+    if (typeof Bun !== 'undefined') return state.buffer.length
```

Comment:
It doesn't seem correct. I think you should use "state.buffer.first().length"

Reply:
I tried testing both solutions with @google-cloud/storage by downloading files from buckets and was unable to get that package working but could be other causes.

The Official Google Cloud Storage Client relies on this package via stream-events -> duplexify -> stream-shift.

Duplexify has 15m+ downloads a week, so hopefully this will be supported soon.

Reply:
Both solutions yield the following error:

`error: incorrect header check
 errno: -3
  code: "Z_DATA_ERROR"

      at node:1:2368
      at node:1:2101
      at node:1:2076
      at node:1:2095
      at node:1:2022`